### DISCUSSION TITLE
Tunnelblick tap devices missing

### DISCUSSION THREAD

**Discussion by andy9a9** [zepp-health/zeppos-forum#246] [Developer Feedback]
Dear developers!

Would it be possible to add some information how to `add tap0..15` via `tunnelblick`? I waste lot of time to get it working. I followed the MAC guide from [tunnelblick installation](https://docs.zepp.com/docs/guides/tools/simulator/setup/#2-configuring-the-virtual-network-card-1), but it isn't enough. Even if you installed the `tun/tap system extension`, devices are not there. It appears only when you add some configuration. Unfortunately the default one has to be modified too. I would say, you should add some basic config there, to get it working.

Hopefully this will help more people 😉.

Andy

**Comment by MyoungXUE** [comment]
Hi, that's a very good suggestion, at this stage our Simulator relies on the tap/tun web extension and the current way of installing it is really cumbersome!

We are working on optimizing the installation method and providing developers with a one-click script to install the environment, which will be released soon along with a documentation update, so stay tuned!
